src/Val.cc

```diff
@@ -3337,39 +3338,49 @@ bool VectorVal::AddTo(Val* val, bool /* is_first_init */) const
 
 const ValPtr& VectorVal::At(unsigned int index) const
 	{
-	if ( index >= val.vector_val->size() )
+	if ( index >= vector_val->size() )
 		return Val::nil;
 
-	return (*val.vector_val)[index];
+	return (*vector_val)[index];
+	}
+
+void VectorVal::Sort(bool cmp_func(const ValPtr& a, const ValPtr& b))
+	{
+	sort(vector_val->begin(), vector_val->end(), cmp_func);
 	}
 
 unsigned int VectorVal::Resize(unsigned int new_num_elements)
 	{
-	unsigned int oldsize = val.vector_val->size();
-	val.vector_val->reserve(new_num_elements);
-	val.vector_val->resize(new_num_elements);
+	unsigned int oldsize = vector_val->size();
+	vector_val->reserve(new_num_elements);
+	vector_val->resize(new_num_elements);
 	return oldsize;
 	}
 
 unsigned int VectorVal::ResizeAtLeast(unsigned int new_num_elements)
 	 {
-	 unsigned int old_size = val.vector_val->size();
+	 unsigned int old_size = vector_val->size();
 	 if ( new_num_elements <= old_size )
 		 return old_size;
 
 	 return Resize(new_num_elements);
 	 }
 
+void VectorVal::Reserve(unsigned int num_elements)
+	{
+	vector_val->reserve(num_elements);
+	}
+
 ValPtr VectorVal::DoClone(CloneState* state)
 	{
 	auto vv = make_intrusive<VectorVal>(GetType<VectorType>());
-	vv->val.vector_val->reserve(val.vector_val->size());
+	vv->Reserve(vector_val->size());
 	state->NewClone(this, vv);
 
-	for ( unsigned int i = 0; i < val.vector_val->size(); ++i )
+	for ( unsigned int i = 0; i < vector_val->size(); ++i )
 		{
-		auto v = (*val.vector_val)[i]->Clone(state);
-		vv->val.vector_val->push_back(std::move(v));
+		auto v = (*vector_val)[i]->Clone(state);
+		vv->Assign(vv->Size(), std::move(v));
```

Comment:
Can you add an `Append()` method to `VectorVal`? It's one of those things I've been meaning to do for a while.